### PR TITLE
Copy on reference tables should never mark placements invalid

### PR DESCRIPTION
**Pull request by onderkalaci**
This commit ensures that COPY does not mark any placement
of reference's state as INVALID in case of an error.

### CODE REVIEW

File: src/backend/distributed/commands/multi_copy.c [hunk 1]
```diff
@@ -598,7 +605,8 @@ CopyToExistingShards(CopyStmt *copyStatement, char *completionTag)
 		/* close the COPY input on all shard placements */
 		EndRemoteCopy(connectionList, true);
 
-		if (MultiShardCommitProtocol == COMMIT_PROTOCOL_2PC)
+		if (MultiShardCommitProtocol == COMMIT_PROTOCOL_2PC ||
+			cacheEntry->replicationModel == REPLICATION_MODEL_2PC)
```
Comment (onderkalaci) [inline src/backend/distributed/commands/multi_copy.c]: Maybe I could move this change into another commit. Do you have any preference?

Comment (marcocitus) [inline src/backend/distributed/commands/multi_copy.c]: @mtuncer I suggest we leave it for the moment. @mtuncer is working on refactoring COPY to use the new connection API. At that point we should call the appropriate 2PC function and remove this code.

**Comment by marcocitus** [comment]
After discussion in person with @onderkalaci this makes sense to me :ship:. The PrepareRemoteTransactions call can be replaced with CoordinatedTransactionUse2PC.
